lib/sentry_dart_plugin.dart

```diff
@@ -275,6 +305,57 @@ class SentryDartPlugin {
     await _executeAndLog('Failed to sources files', params);
   }
 
+  /// Reads available source maps and returns the list of prefixes that should
+  /// be stripped from the source maps. This step is important to make the
+  /// paths clearer and less verbose.
+  ///
+  /// Example 1:
+  /// - expected: `lib/main.dart`
+  /// - actual: `../../../../lib/main.dart`
+  ///
+  /// Example 2:
+  /// - expected: '/flutter/packages/flutter/lib/src/foo.dart'
+  /// - actual '../../SomePath/flutter/packages/flutter/lib/src/foo.dart'
+  Future<Set<String>> _extractPrefixesToStrip(List<File> sourceMapFiles) async {
+    final Set<String> prefixes = {};
+    for (final sourceMapFile in sourceMapFiles) {
+      late final Map<String, dynamic> sourceMap;
+      try {
+        final content = await sourceMapFile.readAsString();
+        sourceMap = jsonDecode(content) as Map<String, dynamic>;
+      } catch (e) {
+        continue;
+      }
+
+      final sources = sourceMap['sources'];
+      if (sources is! List) {
+        continue;
+      }
+
+      final pattern = RegExp(r'^(?:\.\./)+');
+      const flutterFragment = '/flutter/packages/flutter/lib/src/';
+      for (final entry in sources.whereType<String>()) {
+        // Get prefixes for /flutter/packages/flutter/lib/src/
+        // Normalize path separators so the `indexOf` search works on Windows too.
+        final normalised = entry.replaceAll('\\', '/');
```

Comment:
We should also add an additional test with the windows formatting.

Reply:
I'll check whether source maps generated on windows already are normalized to unix forward slashes.

Reply:
@denrase I checked and we don't have to account for windows backslashes, the sourcemaps generation automatically normalizes it. 

see: https://github.com/getsentry/sentry-dart/actions/runs/15134196281/job/42542031629?pr=2953#step:5:209